### PR TITLE
update blacklisted analysis

### LINKED ISSUE #274: Make rita-bl database configurable
Rita creates a database for blacklist called `rita-bl`. This is done [here](https://github.com/activecm/rita/blob/48c009a9bc90b03c60932aed11e3d8e720a14248/analysis/blacklist/blacklist.go#L43).  There is also a static config value [here](https://github.com/activecm/rita/blob/48c009a9bc90b03c60932aed11e3d8e720a14248/config/tables.go#L97) that is ignored due to the above hardcoded value.

Both of these should be replaced and an option placed in the rita config file to allow users to customize the name of this database, if present.  The default name could still be `rita-bl`.  I'd call this extremely low priority as this is the first time this has ever come up and I'm doubtful it's an issue that can't be worked around.

### LINKED ISSUE #304: Update blacklisted analysis

### PR DESCRIPTION
**Pull request by lisaSW**
Closes #304
Closes #274

### CODE REVIEW

**Comment by lisaSW** [comment]
added commit for issue #274

File: config/static_test.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ BlackListed:
     myIP.ms: true
     MalwareDomains.com: true
     MalwareDomainList.com: true
+		BlacklistDatabase: "rita-bl"
```
Comment (ethack) [inline config/static_test.go]: I think this is the spot the test is complaining about.  Just a tab vs. space issue.

Comment (lisaSW) [reply config/static_test.go]: fixed :)

File: analysis/blacklist/analyzer.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	}
+}
+
+// analyze sends a group of timestamps and data sizes in for analysis.
+func (a *analyzerIP) analyzeIP(data *blacklist.IPAnalysisInput) {
+	a.analysisChannel <- data
+}
+
+// analyze sends a group of timestamps and data sizes in for analysis.
+// Note: this function may block
```
Comment (meljbruno) [inline analysis/blacklist/analyzer.go]: May block database reads/writes?

Comment (lisaSW) [reply analysis/blacklist/analyzer.go]: fixed :)

File: analysis/blacklist/blacklist.go [hunk 3]
```diff
@@ -6,49 +6,68 @@ import (
 	"net/http"
 	"os"
 
-	bl "github.com/activecm/rita-bl"
-	blDB "github.com/activecm/rita-bl/database"
+	rita_bl "github.com/activecm/rita-bl"
```
Comment (meljbruno) [inline analysis/blacklist/blacklist.go]: Go lint didn't flag this, but can we change these variables to use mixedCaps instead of underscores?

Comment (lisaSW) [reply analysis/blacklist/blacklist.go]: fixed :)

File: analysis/blacklist/collector.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+				"avg_bytes":   1,
+				"conn_count":  1,
+				"uconn_count": 1,
+				"targets":     1,
+			}},
+		},
+	}
+}
+
+//buildBlacklistedIPs builds a set of blacklisted ips from the
```
[the remark this answers is not in the document]
Comment (lisaSW) [reply analysis/blacklist/collector.go]: yup, it does

File: analysis/blacklist/analyzer.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+func (a *analyzerIP) start() {
+	a.analysisWg.Add(1)
+	go func() {
+		ssn := a.db.Session.Copy()
+		defer ssn.Close()
+
+		for data := range a.analysisChannel {
+
+			var resList []blacklist.RitaBLResult
+			_ = ssn.DB(a.conf.S.Blacklisted.BlacklistDatabase).C("ip").Find(bson.M{"index": data.IP}).All(&resList)
```
[the remark this answers is not in the document]
Comment (lisaSW) [reply analysis/blacklist/analyzer.go]: agreed.

**Pull request merged**